app/models/answer.rb

```diff
@@ -43,22 +44,16 @@ def self.location_answers_for_mission(mission, user = nil, options = {})
     end
 
     # return an AR relation
-    joins(:response, questioning: :question)
-      .joins(%{LEFT JOIN `option_sets` ON `questions`.`option_set_id` = `option_sets`.`id`})
-      .joins(%{LEFT JOIN `options` ON `option_sets`.`allow_coordinates` = 1 AND `questions`.`qtype_name` = 'select_one' AND `answers`.`option_id` = `options`.`id`})
-      .joins(%{LEFT JOIN `choices` ON `option_sets`.`allow_coordinates` = 1 AND `questions`.`qtype_name` = 'select_multiple' AND `answers`.`id` = `choices`.`answer_id`})
-      .joins(%{LEFT JOIN `options` AS `choice_options` ON `choices`.`option_id` = `choice_options`.`id`})
+    joins(:response)
+      .joins(%{LEFT JOIN `choices` ON `choices`.`answer_id` = `answers`.`id`})
       .where(:responses => response_conditions)
       .where(%{
-        (`questions`.`qtype_name` = 'location' AND `answers`.`value` IS NOT NULL)
-        OR (`options`.`latitude` IS NOT NULL AND `options`.`longitude` IS NOT NULL)
-        OR (`choice_options`.`latitude` IS NOT NULL AND `choice_options`.`longitude` IS NOT NULL)
+        (`answers`.`latitude` IS NOT NULL AND `answers`.`longitude` IS NOT NULL)
+        OR (`choices`.`latitude` IS NOT NULL AND `choices`.`longitude` IS NOT NULL)
       })
       .select(:response_id,
-        %{COALESCE(`answers`.`value`,
-                   CONCAT(`options`.`latitude`, ' ', `options`.`longitude`),
-                   CONCAT(`choice_options`.`latitude`, ' ', `choice_options`.`longitude`)) AS `value`})
-      .distinct
+        %{COALESCE(CONCAT(`answers`.`latitude`, ' ', `answers`.`longitude`),
+                   CONCAT(`choices`.`latitude`, ' ', `choices`.`longitude`)) AS `value`})
```

Comment:
I'm not a huge fan of this one piece. I see that it is required in order to deprecate smoothly. So I guess it stays, but could we at least put a comment on it saying that it's temporary and perhaps listing the places in the app that depend on it.

Could we also refactor the dashboard map to use the new lat/lng cols instead of this? Shouldn't be hard. I think even that that code might be doing needless joins right now since it hasn't been touched at all in this PR.